packages/expo-camera2/src/ExpoCamera2View.tsx

```diff
@@ -0,0 +1,275 @@
+   * Photos are rotated to match device's orientation (on Android only if **options.skipProcessing** flag is not enabled) and scaled to match the preview.
+   * TODO: Anyway on Android it is essential to set ratio prop to get a picture with correct dimensions.
+   *
+   * > Resulting local image URI is temporary.
+   * > Use **Expo.FileSystem.copyAsync** to make a permanent copy of the image.
+   */
+  @ensureAvailable
+  takePictureAsync(options: TakingPictureOptions = {}): Promise<Picture> {
+    return ExpoCamera2ViewManager.takePictureAsync(options, this.cameraNodeHandle);
+  }
+
+  /**
+   * Starts recording a video that will be saved to cache directory.
+   * Videos are rotated to match device's orientation.
+   * Resulting promise is returned either if:
+   * - **stopRecording** method was invoked
+   * - one of **maxDuration** aor **maxFileSize** is reached
+   * - TODO: or camera preview is stopped (what about **pausePreviewAsync** method?)
+   *
+   * TODO: Flipping camera during a recording results in stopping it?
+   *
+   * > Resulting video URI is temporary.
+   * > Use **Expo.FileSystem.copyAsync** to make a permanent copy of the video.
+   */
+  @ensureAvailable
+  recordAsync(options: VideoRecordingOptions = {}): Promise<Video> {
+    return ExpoCamera2ViewManager.recordAsync(options, this.cameraNodeHandle);
+  }
+
+  /**
+   * Stops recording if any is in progress.
+   */
+  @ensureAvailable
+  stopRecordingAsync(): Promise<void> {
+    return ExpoCamera2ViewManager.stopRecordingAsync(this.cameraNodeHandle);
+  }
+
+  /**
+   * Get aspect ratios that are supported by the device.
+   * @Android only
+   */
+  @ensureAvailable
+  getAvailableRatiosAsync(): Promise<string[]> {
+    return ExpoCamera2ViewManager.getAvailableRatiosAsync(this.cameraNodeHandle);
+  }
+
+  /**
+   * Get picture sizes that are supported by the device for given ratio.
+   * Returned list varies across **Android** devices but is the same for every **iOS**.
+   */
+  @ensureAvailable
+  getAvailablePictureSizesAsync(ratio: string): Promise<string[]> {
+    return ExpoCamera2ViewManager.getAvailablePictureSizesAsync(ratio, this.cameraNodeHandle);
+  }
+
+  /**
+   * Tries to focus camera device on given point.
+   * Efect is temporary and upon any device movement might be discarded.
+   */
+  @ensureAvailable
+  focusOnPoint(previewFocusPoint: FocusPoint): Promise<boolean> {
+    return ExpoCamera2ViewManager.focusOnPoint(previewFocusPoint, this.cameraNodeHandle);
+  }
+
+  setNativeReference(ref: React.Component | null) {
+    if (ref) {
+      const cameraNodeHandle = findNodeHandle(ref)
+      if (cameraNodeHandle) {
+        this.cameraNodeHandle = cameraNodeHandle
+      }
+    } else {
+      this.cameraNodeHandle = undefined;
+    }
+  }
+
+  render() {
+    return (
+      <ExpoNativeCameraView
+        {...this.props}
+        ref={this.setNativeReference}
+      />
+    )
+  }
+}
+
+function ensureAvailable<T extends Function>(
```

Comment:
Why do we need something like that?

Reply:
Well, most modules have some mechanism checking out whether method is available on specific platform - see any module that supports web. I wanted one clear way for ensuring method is available from native side and here's my neat way for doing it. Long live decorators! 😄